CMakeLists.txt

```diff
@@ -27,8 +27,20 @@ IF (POLICY CMP0074)
   CMAKE_POLICY(SET CMP0074 NEW)
 ENDIF ()
 
+SET (MQTT_TLS_OPENSSL "MQTT_TLS_OPENSSL")
+SET (MQTT_TLS_GNUTLS "MQTT_TLS_GNUTLS")
+
 IF (MQTT_USE_TLS)
     MESSAGE (STATUS "TLS enabled")
+    IF (MQTT_USE_TLS STREQUAL ${MQTT_TLS_GNUTLS})
+        MESSAGE (STATUS "Using GNUTLS")
+        IF (MQTT_BOOST_ASIO_GNUTLS_INCLUDE STREQUAL "")
```

Comment:
Why namespace the include directory like this?

If you need the ability to insert additional include directories, just name it "MQTT_ADDITIONAL_INCLUDE" or something, and take list as the argument.

Are you sure CMake doesn't already have a specific feature for this built in?

Reply:
Couldn't find a native Cmake way of doing that.

As per this [stack overflow post](https://stackoverflow.com/a/35633733/2036016), the other option we have is to let consumers set CXX flags which doesn't seem too appealing. 

I can change it to be MQTT_ADDITIONAL_INCLUDES and accept a list if that works

Reply:
Personally I don't see the benefit of making something that clearly applies in a generic way named in a feature specific way.

I think it'll cause more confusion to people to have it named MQTT_BOOST_ASIO_GNUTLS_INCLUDE.

@redboltz can you offer some guidance?

Reply:
I got confused. I don't understand what you want to do, so far.
Do you simply want to add include path of GNUTLS ? 
If it does, `-DCMAKE_CXX_FLAGS+="-I path_to_gnutls"`.